webapp/src/components/todo_item/todo_item.jsx

```diff
@@ -0,0 +1,337 @@
+
+import CompleteButton from '../buttons/complete';
+import AcceptButton from '../buttons/accept';
+import {
+    canComplete,
+    canRemove,
+    canAccept,
+    canBump,
+    handleFormattedTextClick,
+} from '../../utils';
+import CompassIcon from '../icons/compassIcons';
+import Menu from '../../widget/menu';
+import MenuItem from '../../widget/menuItem';
+import MenuWrapper from '../../widget/menuWrapper';
+import Button from '../../widget/buttons/button';
+
+const PostUtils = window.PostUtils; // import the post utilities
+
+function TodoItem(props) {
+    const {issue, theme, siteURL, accept, complete, list, remove, bump, openTodoToast, openAssigneeModal, setEditingTodo, editIssue} = props;
+    const [done, setDone] = useState(false);
+    const [editTodo, setEditTodo] = useState(false);
+    const [message, setMessage] = useState(issue.message);
+    const [description, setDescription] = useState(issue.description);
+    const [hidden, setHidden] = useState(false);
+
+    const style = getStyle(theme);
+
+    const handleClick = (e) => handleFormattedTextClick(e);
+
+    const htmlFormattedMessage = PostUtils.formatText(issue.message, {
+        siteURL,
+    });
+
+    const htmlFormattedDescription = PostUtils.formatText(issue.description, {
+        siteURL,
+    });
+
+    const issueMessage = PostUtils.messageHtmlToComponent(htmlFormattedMessage);
+    const issueDescription = PostUtils.messageHtmlToComponent(htmlFormattedDescription);
+
+    let listPositionMessage = '';
+    if (issue.user) {
+        if (issue.list === '') {
+            listPositionMessage =
+                'Accepted. On position ' + (issue.position + 1) + '.';
+        } else if (issue.list === 'in') {
+            listPositionMessage =
+                'In Inbox on position ' + (issue.position + 1) + '.';
+        } else if (issue.list === 'out') {
+            listPositionMessage = '';
+        }
+    }
+
+    const listDiv = (
+        <div
+            className='light'
+            style={style.subtitle}
+        >
+            {listPositionMessage}
+        </div>
+    );
+
+    const acceptButton = (
+        <AcceptButton
+            issueId={issue.id}
+            accept={accept}
+        />
+    );
+
+    const onKeyDown = (e) => {
+        if (e.key === 'Enter') {
+            saveEditedTodo();
+        }
+
+        if (e.key === 'Escape') {
+            setEditTodo(false);
+        }
+    };
+
+    const actionButtons = (
+        <div className='todo-action-buttons'>
+            {canAccept(list) && acceptButton}
+        </div>
+    );
+
+    let removeTimeout;
+    const completeTimeout = useRef(null);
+
+    const completeToast = useCallback(() => {
+        openTodoToast({icon: 'check', message: 'Todo completed', undo: undoCompleteTodo});
+
+        setHidden(true);
+
+        completeTimeout.current = setTimeout(() => {
+            complete(issue.id);
+        }, 5000);
+    }, [complete, openTodoToast, issue, completeTimeout]);
```

Comment:
completeTimeout is always the same between renders (because of how `useRef` works) so there is no need to have it in the dependencies.